shuup/core/models/_products.py

```diff
@@ -568,7 +571,13 @@ def soft_delete(self, user=None):
             # Bypassing local `save()` on purpose.
             super(Product, self).save(update_fields=("deleted",))
 
+    # TODO (2.0): deprecate this
     def verify_mode(self):
+        warnings.warn(
```

Comment:
First, it's wrongly named it should be `set_mode` or similar.

Real world usage
As I was making my addon, I realized that my addon cannot set the `ProductMode` for `Product` as it was always overridden with this magical method. So, this method is not extendable. I created a hacked solution for this in my addon, but it's far from ideal solution imo.

Proposals for 2.0 (at least):
* This method should be renamed
* This method should set the `ProductMode` based on business logic (I actually had a solution for this, but didn't want to add too much in this PR). The solution is to have a set of `ProductMode` setters which can be provided through `provides`. Pretty simple, but effective.
* `ProductMode` should not be `Enum` anymore, but something that could be extended from Addons otherwise we would have to figure out all the possible product modes in the world :)